traitsui/extras/demo.py

```diff
@@ -523,17 +549,23 @@ class DemoPath(DemoTreeNodeObject):
     parent = Any()
 
     #: Name of file system path to this package:
-    path = Property()
+    path = Property(depends_on='parent.path,name')
+
+    #: Description of what the demo does:
+    description = HTML()
```

Comment:
This does need to be a property in order to exercise `_get_description` and get the cover image loaded.
```suggestion
    description = Property(HTML, depends_on="_description")
```